mitosheet/mitosheet/public/v3/formatting.py

```diff
@@ -0,0 +1,36 @@
+from openpyxl import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+
+from openpyxl.styles import Font, PatternFill
+from openpyxl.styles import NamedStyle
+
+def add_formatting_to_excel_sheet( params: dict[Workbook, str, dict]) -> None:
```

Comment:
Let's make the parameters explicitly named variables to the function. A good format sounds like:

```
def add_formatting_to_excel_sheet(
     writer: pd.ExcelWriter, 
     sheet_name: str,
     header_background_color: Optional[str]=None,
     header_color: Optional[str]=None,
):
```
A few notes on the params: 
1. It's taking a writer, rather than the book (I think this is a clearer interface - and what one expects).
2. It takes all the specific formatting variables as explicitly arguments. 

The approach you took is very react (e.g. everything is wrapper in props), and results in something that is easy to call with our current params object. If we move to this new version I propose, it's a bit harder to call, but we get the benefits of a strongly typed function and a _very clear_ public interface. Having a very clear public interface is important as this is exposed as a public function to users!

We'll also have to do a bit more work to transpile the function params - but it's not too bad. You can see utilities for doing that in `excel_range_import_code_chunk.py` file -- peep the `param_dict_to_code` helper.